docker/Dockerfile.gpu

```diff
@@ -0,0 +1,149 @@
+    
+    apt-get update && \
+    #apt-get install cuda-10-1 -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+RUN add-apt-repository universe && apt-get update &&  apt-get install -y jq
+
+RUN curl -O http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/cuda-repo-ubuntu1604_10.1.168-1_amd64.deb && \
+    dpkg --force-confdef --force-confnew -i ./cuda-repo-ubuntu1604_10.1.168-1_amd64.deb && \
+    apt-key adv --fetch-keys http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/7fa2af80.pub && \
+   
+    curl -O http://ftp.ubuntu.com/ubuntu/ubuntu/pool/restricted/n/nvidia-graphics-drivers-418/libnvidia-compute-418_418.56-0ubuntu1_amd64.deb && \
+    dpkg --force-confdef --force-confnew -i ./libnvidia-compute-418_418.56-0ubuntu1_amd64.deb && \
+
+    curl -O http://ftp.ubuntu.com/ubuntu/ubuntu/pool/restricted/n/nvidia-graphics-drivers-418/libnvidia-decode-418_418.56-0ubuntu1_amd64.deb && \
+    dpkg --force-confdef --force-confnew -i ./libnvidia-decode-418_418.56-0ubuntu1_amd64.deb && \
+
+    curl -O http://ftp.ubuntu.com/ubuntu/ubuntu/pool/restricted/n/nvidia-graphics-drivers-418/libnvidia-encode-418_418.56-0ubuntu1_amd64.deb && \
+    dpkg --force-confdef --force-confnew -i ./libnvidia-encode-418_418.56-0ubuntu1_amd64.deb
+
+# everything above this line has already been cached in the build stage -
+# we save time and disk space by keeping the common parts of each stage
+# identical at the start of each image build
+
+COPY --from=builder /usr/local/lib/* /usr/local/lib/
+COPY --from=builder /root/go/src/github.com/livepeer/go-livepeer/devtool /usr/local/bin/
+COPY --from=builder /root/go/src/github.com/livepeer/go-livepeer/livepeer /usr/local/bin/
+
+RUN ln -s /usr/local/bin/livepeer /
+RUN ln -s /usr/local/bin/livepeer /usr/bin/
+ENV LD_LIBRARY_PATH /usr/local/lib/
+
+# Keylase patch
+ENV NVIDIA_VISIBLE_DEVICES all
+ENV NVIDIA_DRIVER_CAPABILITIES compute,video,utility
+
+RUN mkdir -p /usr/local/bin /patched-lib
+COPY docker/patch.sh docker/gpu-start.sh /usr/local/bin/
+RUN chmod +x /usr/local/bin/patch.sh /usr/local/bin/gpu-start.sh
+
+
+#ENTRYPOINT [ "/root/go/bin/livepeer" ]
+WORKDIR /root
+
+#RUN apt-get install -y libnvidia-encode-418   
+
+RUN mkdir -p /root/.lpData/mainnet/keystore && \
+  mkdir -p /root/.lpData/rinkeby/keystore && \
+  mkdir -p /root/.lpData/devenv/keystore && mkdir -p /root/.lpData/offchain/keystore
+#RUN cp /root/go/src/github.com/livepeer/go-livepeer/livepeer /usr/bin/livepeer
+#RUN cp /root/go/src/github.com/livepeer/go-livepeer/livepeer_cli /usr/bin/livepeer_cli
+
+#COPY --from=builder /go/src/github.com/livepeer/go-livepeer/livepeer /usr/bin/livepeer
+#COPY --from=builder /go/src/github.com/livepeer/go-livepeer/livepeer_cli /usr/bin/livepeer_cli
+
+#COPY docker/start.sh .
+#RUN chmod +x start.sh
+
+EXPOSE 7935/tcp
+EXPOSE 8935/tcp
+EXPOSE 1935/tcp
+
+ENTRYPOINT ["/usr/local/bin/gpu-start.sh"]
+CMD ["--help"]
+
```

Comment:
I think this this should all be done in yet another stage.   Perhaps the "final" stage should be called "intermediate" and this would be the "final" stage.   I don't feel strongly about this, but it does feel like a lot of changes related to configuration rather than image building.